tests/qa_automation/qa_run.pm

```diff
@@ -109,20 +108,17 @@ sub qa_script_output {
 
 # Check whether DONE file exists every $interval secs in the background
 sub wait_testrun {
-    my ($self, $interval) = @_;
-    $interval = 30 unless defined $interval;
+    my $self      = shift;
+    my $interval  = shift || 30;
     my $done_file = '/var/log/qaset/control/DONE';
     my $pattern   = "TESTRUN_FINISHED-" . int(rand(999999));
-    my $cmd       = "while [[ ! -f $done_file ]]; do sleep $interval; done; echo $pattern >> /dev/$serialdev";
-    type_string "bash -c '$cmd' &\n";
+    script_run("while [[ ! -f $done_file ]]; do sleep $interval; done; echo $pattern >> /dev/$serialdev", 0);
     # Set a high timeout value for wait_serial
     # so that it will wait until test run finished or
     # MAX_JOB_TIME(can be set on openQA webui) reached
     my $ret = wait_serial($pattern, 180 * 60);
-    if ($ret) {
-        return 1;
-    }
-    return 0;
+    return 1 if $ret;
+    return undef;
```

Comment:
'return undef' is the same as 'return;' but longer. but in this very case I would suggest 'return $ret ? 1 : 0;'